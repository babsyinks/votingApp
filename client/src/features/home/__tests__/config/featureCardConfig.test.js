import React from "react";
import items from "features/home/config/featureCardConfig";

describe("featureCardConfig", () => {
  it("should export an object with 'expect' and 'features'", () => {
    expect(items).toHaveProperty("expect");
    expect(items).toHaveProperty("features");
  });

  it("each section should have a title and contents array", () => {
    Object.values(items).forEach((section) => {
      expect(typeof section.title).toBe("string");
      expect(Array.isArray(section.contents)).toBe(true);
      expect(section.contents.length).toBeGreaterThan(0);
    });
  });

  it("each content item should have id, icon, title, and description", () => {
    Object.values(items).forEach((section) => {
      section.contents.forEach((item) => {
        expect(typeof item.id).toBe("string");
        expect(item.id.length).toBeGreaterThan(0);

        const Icon = item.icon;
        expect(Icon).toBeTruthy();
        expect(React.isValidElement(<Icon />)).toBe(true);

        expect(typeof item.title).toBe("string");
        expect(item.title.length).toBeGreaterThan(0);
        expect(typeof item.description).toBe("string");
        expect(item.description.length).toBeGreaterThan(0);
      });
    });
  });

  it("should generate unique ids for each content item", () => {
    const allIds = [
      ...items.expect.contents.map((c) => c.id),
      ...items.features.contents.map((c) => c.id),
    ];
    expect(new Set(allIds).size).toBe(allIds.length);
  });
});
