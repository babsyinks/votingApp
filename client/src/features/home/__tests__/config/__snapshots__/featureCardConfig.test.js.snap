// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`featureCardConfig should match the expected structure (snapshot) 1`] = `
Object {
  "expect": Object {
    "contents": ArrayContaining [
      ObjectContaining {
        "id": Any<String>,
      },
    ],
    "title": "What You Can Expect",
  },
  "features": Object {
    "contents": ArrayContaining [
      ObjectContaining {
        "id": Any<String>,
      },
    ],
    "title": "Features of VoteNow",
  },
}
`;
